### PR TITLE
Profiler remotery

### LINKED ISSUE #6394: Stream profiler data using a socket instead of providing it by http requests
**Is your feature request related to a problem? Please describe (REQUIRED):**
Sometimes it's hard to catch exact frame where problem appears using current profiler implementation.
Streaming the profiling data constantly would help to resolve the issue.

**Describe the solution you'd like (REQUIRED):**
That would be great to be able to connect to the socket connection in the engine and receive data from profiler.
Having documentation for the data format is also important in this case

### PR DESCRIPTION
**Pull request by JCash**
The custom profiler used in Defold has been replaced with the [Remotery](https://github.com/Celtoys/Remotery) profiler. The new profiler will stream profiler data instead of sampling only 20 frames.

Fixes #6394

### CODE REVIEW

File: editor/src/clj/editor/engine_profiler.clj [hunk 1]
```diff
@@ -0,0 +1,76 @@
+;; Copyright 2020-2022 The Defold Foundation
```
Comment (JCash) [inline editor/src/clj/editor/engine_profiler.clj:1]: I made a copy of the existing `web_profiler.clj`

File: engine/dlib/src/dlib/profile/profile.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+#if defined(__ANDROID__)
+    #define PROFILE_FMT_U64 "%llu"
+    #define PROFILE_FMT_I64 "%lld"
+#else
+    #include <inttypes.h>
+    #define PROFILE_FMT_U64 "%" PRIu64
+    #define PROFILE_FMT_I64 "%" PRId64
+#endif
```
Comment (JCash) [inline engine/dlib/src/dlib/profile/profile.cpp:36]: Should probably be moved to dstrings.h later on.

File: engine/dlib/src/dmsdk/dlib/profile.h [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+    #define DM_PROFILE_DYN(name, hash_cache) \
+        _DM_PROFILE_SCOPE(name, hash_cache)
+
+    #define DM_PROFILE_TEXT_LENGTH 1024
+    #define DM_PROFILE_TEXT(format, ...)              dmProfile::LogText(format, __VA_ARGS__)
+
+    #define DM_COUNTER_DYN(counter_index, amount)
+
+    // The profiler property api
+    #define DM_PROPERTY_EXTERN(name)                                rmt_PropertyExtern(name)
```
Comment (JCash) [inline engine/dlib/src/dmsdk/dlib/profile.h:554]: The new profile property API

File: engine/dlib/src/dmsdk/external/remotery/Remotery.h [hunk 4]
```diff
@@ -0,0 +1,1089 @@
+
```
Comment (JCash) [inline engine/dlib/src/dmsdk/external/remotery/Remotery.h:1]: Due to the fact that the Remotery api uses defines, we need to have this header in dmsdk. But it is to be considered private.

File: engine/dlib/src/remotery/update_remotery.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
```
Comment (JCash) [inline engine/dlib/src/remotery/update_remotery.sh:1]: Followind our upgrade guidelines in readme.md, this script makes it easy to update to the latest version of Remotery

File: share/extender/variants/release.appmanifest [hunk 6]
```diff
@@ -6,47 +6,47 @@ platforms:
 
     x86_64-osx:
         context:
-            excludeLibs: ["engine", "engine_service", "profilerext", "record", "vpx"]
-            libs: ["engine_release", "engine_service_null", "profilerext_null", "record_null"]
+            excludeLibs: ["engine", "engine_service", "profile", "remotery", "profilerext", "record", "vpx"]
+            libs: ["engine_release", "engine_service_null", "profile_null", "remotery_null", "profilerext_null", "record_null"]
```
Comment (JCash) [inline share/extender/variants/release.appmanifest:10]: Let's not link with the profiler+remotery by default in release builds.

**Review by britzl — DISMISSED** [review]

**Comment by AGulev** [comment]
Fix https://github.com/defold/defold/issues/4738 https://github.com/defold/defold/issues/6600

**Pull request merged**